src/test/java/org/gridsuite/filter/server/expertrule/EnumExpertRuleTest.java

```diff
@@ -0,0 +1,167 @@
+package org.gridsuite.filter.server.expertrule;
+
+import com.powsybl.iidm.network.*;
+import org.gridsuite.filter.server.dto.expertfilter.expertrule.EnumExpertRule;
+import org.gridsuite.filter.server.utils.expertfilter.FieldType;
+import org.gridsuite.filter.server.utils.expertfilter.OperatorType;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mockito;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static org.gridsuite.filter.server.utils.expertfilter.OperatorType.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class EnumExpertRuleTest {
+
+    @ParameterizedTest
+    @MethodSource({
+        "provideArgumentsForGeneratorTest",
+        "provideArgumentsForLoadTest",
+        "provideArgumentsForBusTest",
+        "provideArgumentsForBusBarSectionTest"})
+    void testEvaluateRule(OperatorType operator, FieldType field, String value, Set<String> values, Identifiable<?> equipment, boolean expected) {
+        EnumExpertRule rule = EnumExpertRule.builder().operator(operator).field(field).value(value).values(values).build();
+        assertEquals(expected, rule.evaluateRule(equipment));
+    }
+
+    private static Stream<Arguments> provideArgumentsForGeneratorTest() {
+
+        Generator gen = Mockito.mock(Generator.class);
+        Mockito.when(gen.getType()).thenReturn(IdentifiableType.GENERATOR);
+        // Generator fields
+        Mockito.when(gen.getEnergySource()).thenReturn(EnergySource.HYDRO);
+        // VoltageLevel fields
+        Substation substation = Mockito.mock(Substation.class);
+        VoltageLevel voltageLevel = Mockito.mock(VoltageLevel.class);
+        Mockito.when(voltageLevel.getSubstation()).thenReturn(Optional.of(substation));
+        Terminal terminal = Mockito.mock(Terminal.class);
+        Mockito.when(terminal.getVoltageLevel()).thenReturn(voltageLevel);
+        Mockito.when(gen.getTerminal()).thenReturn(terminal);
+        Mockito.when(substation.getCountry()).thenReturn(Optional.of(Country.FR));
+
+        return Stream.of(
+                // --- EQUALS --- //
+                // Generator fields
+                Arguments.of(EQUALS, FieldType.ENERGY_SOURCE, EnergySource.HYDRO.name(), null, gen, true),
```

Comment:
Same as above, need to test `true` and `false`, otherwise we don't really test it

Reply:
Still missing

Reply:
DONE by https://github.com/gridsuite/filter-server/pull/90/commits/0fef89ec3a4c457b5aa7e85f8169ade69acafe1c